src/EFCore.SqlServer/Update/Internal/SqlServerUpdateSqlGenerator.cs

```diff
@@ -767,12 +767,15 @@ protected virtual int MergeIntoMinimumThreshold
     private static string GetTypeNameForCopy(IProperty property)
     {
         var typeName = property.GetColumnType();
+        var collation = property[RelationalAnnotationNames.Collation]?.ToString();
 
-        return property.ClrType == typeof(byte[])
+        typeName = property.ClrType == typeof(byte[])
             && (typeName.Equals("rowversion", StringComparison.OrdinalIgnoreCase)
                 || typeName.Equals("timestamp", StringComparison.OrdinalIgnoreCase))
                 ? property.IsNullable ? "varbinary(8)" : "binary(8)"
                 : typeName;
+
+        return collation is not null ? $"{typeName} COLLATE {collation}" : typeName;
```

Comment:
Reverse condition to avoid double negation
```suggestion
        return collation is null ? typeName : $"{typeName} COLLATE {collation}";
```